### ISSUE TITLE
Case splits on tuples

### ISSUE THREAD

**Issue by gpyh** [idris-lang/Idris-dev#3166]
Say I have a type `(Int, Int, Int)`. If a case split on a variable of this type, I get `(a, b)`. I'd like to have `(a, b, c)` instead (although tuples are implemented as nested pairs).

Tested with Atom and Vim.

**Comment by mdimjasevic** [comment]
Just to add some clarification what is happening here.

When you provide Idris with the following program:

```Idris
fun : (t : (Int, Int, Int)) -> Int
fun t = ?fun_rhs
```

and then case split on `t`, Idris gives this:

```idris
fun : (t : (Int, Int, Int)) -> Int
fun (a, b) = ?fun_rhs_1
```

Here, `a` is of type `Int`, and `b` is of type `(Int, Int)`.

Is this desired behavior? Or would the following be desired:

```idris
fun : (t : (Int, Int, Int)) -> Int
fun (a, b, c) = ?fun_rhs_1
```

where `a`, `b` and `c` would all be of type `Int`?

**Comment by ahmadsalim** [comment]
@mdimjasevic I agree that the second behaviour where it splits to `(a,b,c)` directly would be the more intuitive one.

**Comment by mdimjasevic** [comment]
@ahmadsalim , thanks for commenting.

And how about this case?

```idris
tup : (t : (Int, (Bool, Nat))) -> Bool
tup t = ?tup_rhs
```

Do we want option 1 or option 2?

Option 1:
```idris
tup : (t : (Int, (Bool, Nat))) -> Bool
tup (a, b) = ?tup_rhs_1
```
where `a : Int` and `b : (Bool, Nat)`.

Option 2:
```idris
tup : (t : (Int, (Bool, Nat))) -> Bool
tup (a, (b, c)) = ?tup_rhs_1
```
where `a : Int`, `b : Bool` and `c : Nat`.

Based on what @ahmadsalim said above, I guess option 1 is desired, i.e. expanding only at the very first level of a tuple without going to deeper nested levels. If that is the case, then we'd have to keep track of how the user wrote it in the function type, and not necessarily rely on how Idris represent a tuple internally.

**Comment by ahmadsalim** [comment]
@mdimjasevic I would also say that option 1 is desired here yes, but I am unsure if the representation internally is precise enough to make a distinction between `(A,B,C)` and `(A, (B,C))` in the types. If you can only get one solution working, I think expanding as much as possible is preferred.
